include/uk/init.h

```diff
@@ -55,88 +79,117 @@ typedef int (*uk_init_func_t)(void);
  *   Use the UK_PRIO_AFTER() helper macro for computing priority dependencies.
  *   Note: Any other value for level will be ignored
  */
-#define __UK_INITTAB(fn, base, prio)					\
-	static const uk_init_func_t					\
-	__used __section(".uk_inittab" #base #prio)			\
-	__uk_inittab ## base ## prio ## _ ## fn = (fn)
+#define __UK_INITTAB_ENTRY(init_fn, term_fn, base, prio)		\
+	static const struct uk_inittab_entry				\
+	__used __section(".uk_inittab" #base #prio) __align(8)		\
+		__uk_inittab ## base ## prio ## _ ## init_fn ## _ ## term_fn = {\
+		.init = (init_fn),					\
+		.term = (term_fn)					\
+	}
 
-#define _UK_INITTAB(fn, base, prio)					\
```

Comment:
I would prefer to not change the existing interface and instead introduce a new macro that allows specifying the `term_fn`. Then it wouldn't break all existing (external) clients of the API.

Reply:
The update requires a macro that sets both at the same time because this is the way how the table is constructed, the same priority value is used for initialization and termination (reverse order). This simplifies cleaning up in case of errors of one of the init function.
Of course we could argue, another macro could be provided as wrapper that sets only init but I wanted to have this explicitly. The fact that the system can shutdown should also make developers think about a termination. As soon this is upstreamed, we have a number of PR across libs that fix existing code so that disruption should be minimal.